### ISSUE TITLE
R-CNN FG Accuracy=0.000000

### ISSUE THREAD

**Issue by SingL3** [msracver/Deformable-ConvNets#202]
Hello! I try a detection model and I get R-CNN FG Accuracy=0.000000. Is it ok if I just keep running it?
I search about this and it is said that I have to adjust the base_lr for gpu numbers. I do that but still get this issue.
Any suggestions?
Thank you!

**Comment by huliang2016** [comment]
maybe you can try reduce learning rate

**Comment by maozezhong** [comment]
@SingL3 hello, have you solved this problem? Please let me know, Thanks adavance

**Comment by SingL3** [comment]
@maozezhong I do as what he suggests, reduce the learning rate to adjust the number of gpus you are using. Originally, for coco dataset, it sets to 0.01 for 8 gpus. I use 4 gpus so I set to 0.005.

**Comment by SingL3** [comment]
R-CNN FG Accuracy=0.000000
This will still occur for hundreds of batches or thousands of batches, but after that it will increase.

**Comment by maozezhong** [comment]
OK. thank you for your reply. After 3000 batches, it increase to 0.000055, still so small...
I will continue training and see what it will increase to.
I have another question, what acturally does this 'FG' means ?

**Comment by SingL3** [comment]
@maozezhong When it is not 0.0000000, I will increase constantly. I think FG means foreground. However, I found that this metric is kind of useless. I test my model at epoch 2 when it is ~0.5 and epoch 7 when it is ~0.82, but the performance of epoch 2 is better than 7.

**Comment by maozezhong** [comment]
OK, I got it, Thanks!

**Comment by yhs95** [comment]
All the above answers are wrong. Accuracy = 0 is not due to lr.That's caused by batch_rois=-1, which is in the experiments/fpn/cfgs/resnet_v1_101_coco_trainval_fpn_dcn_end2end_ohem.yaml file.
Change batch_rois=-1 to batch_rios=512.

**Comment by yhs95** [comment]
All the above answers are wrong. Accuracy = 0.000000 is not due to lr.That's caused by batch_rois=-1, which is in the experiments/fpn/cfgs/resnet_v1_101_coco_trainval_fpn_dcn_end2end_ohem.yaml file.
Change batch_rois=-1 to batch_rios=512.

**Comment by SingL3** [comment]
@yhs95 This hyper-parameter is set to -1 for using ohem, but actually I didn't use ohem and I still get this issue at that time.

**Comment by yhs95** [comment]
Is wrong.This hyper-parameter is not set for using ohem,It is used to select rios. After I modified this parameter, the problem of Accuracy = 0.0000 was solved. After testing, the AP value returned to normal.

**Comment by yhs95** [comment]
Epoch[1] Batch [1200]	Speed: 1.70 samples/sec	Train-RPNAcc=0.974165,	RPNLogLoss=0.069415,	RPNL1Loss=0.067423,	Proposal FG Fraction=0.137024,	R-CNN FG Accuracy=0.370897,	RCNNAcc=0.889737,	RCNNLogLoss=0.334114.
you can result for output of training.

**Comment by yhs95** [comment]
Epoch[1] Batch [1200] Speed: 1.70 samples/sec Train-RPNAcc=0.974165, RPNLogLoss=0.069415, RPNL1Loss=0.067423, Proposal FG Fraction=0.137024, R-CNN FG Accuracy=0.370897, RCNNAcc=0.889737, RCNNLogLoss=0.334114.
you can result for output of training,it is obvious that 'accuracy' already is normal value.

**Comment by yhs95** [comment]
Epoch[1] Batch [1200] Speed: 1.70 samples/sec Train-RPNAcc=0.974165, RPNLogLoss=0.069415, RPNL1Loss=0.067423, Proposal FG Fraction=0.137024, R-CNN FG Accuracy=0.370897, RCNNAcc=0.889737, RCNNLogLoss=0.334114.you can see result for output of my  training,it is obvious that 'accuracy' already is normal value.
